source/rock/middle/TypeDecl.ooc

```diff
@@ -463,6 +464,43 @@ TypeDecl: abstract class extends Declaration {
 
     }
 
+    checkFinalInherit: func(res: Resolver) -> Bool{
+        list := ArrayList<TypeDecl> new()
+        current := this
+
+        while(current != null) {
+            if(current getSuperType() == null) break // it's alright
+
+            next := current getSuperRef()
+            if(next == null) {
+                res wholeAgain(this, "need superRef to check final redefine")
+                return false
+            }
+
+            list add(current)
+            current = next
+        }
+
+        if(list size > 2){
+            for(i in 0..list size - 1){
+                for(j in i+1..list size){
+                    list[i] functions each(|fdecl|
+                        if(fdecl name == "init" || fdecl name == "new"){ return }
```

Comment:
I definitely agree that `init: final func` should be disallowed. We can have a 'final class' or 'final func' but never a 'final constructor'. 

My point is that we should throw an error when parsing this rather than in resolve stage.  `Can not override a final function "init"` is a really wired(misleading) message, because this will make people think that they can not use final init just because they are trying to override it but not they are using the combination "`final` and `init`". 

I will make more patches to front/middle end to make 'final class' work and disable final init. After that we can delete this safely.